metabase/metabase
# Hide unmapped filters for public/embedded dashboards

Followup on #37711 (issue #36359)

After recent changes, we're now hiding a dashboard filter when it's not mapped to at least one card on the current tab
This was meant to work the same for public and statically embedded dashboards, but I missed that this code isn't shared between public/embedded and in-app dashboards.

### To verify

1. Sign in as admin
2. Go to `/admin/settings/public-sharing` and enable public sharing
3. Create a dashboard with a couple of tabs, several filters, and dashboards. Ensure you have a filter mapped to at least one card per tab, a filter used by a single tab, and a filter not used by any card at all.
4. While the dashboard page, click the sharing icon (in the top right), generate a public link and open it
5. Click through tabs, ensure filter are hidden on tabs where they're not connected to any cards
